main.cpp

```diff
@@ -395,29 +197,29 @@ int main()
     // FloatType object instanciation and method tests
     FloatType ft (3.0f);
     // --------
-    std::cout << "Initial value of ft: " << *(ft.value) << std::endl;
+    std::cout << "Initial value of ft: " << float(ft) << std::endl;
     // --------
     std::cout << "Use of function concatenation (only float arguments) " << std::endl;
-    std::cout << "New value of ft = (ft + 3.0f) * 1.5f / 5.0f = " << *(ft.add( 3.0f ).multiply(1.5f).divide(5.0f).value) << std::endl;
+    std::cout << "New value of ft = (ft + 3.0f) * 1.5f / 5.0f = " << float( (ft.add( 3.0f ).multiply(1.5f).divide(5.0f)) ) << std::endl;
        
     std::cout << "---------------------\n" << std::endl; 
     
     // DoubleType/IntType object instanciation and method tests
     DoubleType dt (5.0);
     IntType it(10);
     // --------
-    std::cout << "Initial value of dt: " << *(dt.value) << std::endl;
-    std::cout << "Initial value of it: " << *(it.value) << std::endl;
+    std::cout << "Initial value of dt: " << double(dt) << std::endl;
+    std::cout << "Initial value of it: " << int(it) << std::endl;
     // --------
     std::cout << "Use of function concatenation (mixed type arguments) " << std::endl;
-    std::cout << "New value of dt = (dt * it) / 5.0f + ft = " << *(dt.multiply(it).divide(5.0f).add(ft).value) << std::endl;
+    std::cout << "New value of dt = (dt * it) / 5.0f + ft = " << double( dt.multiply(it).divide(double(5.0f)).add(double(ft)) ) << std::endl;
```

Comment:
FIXME this cast should be unnecessary now that you have User-Defined Conversion functions in place
- [ ] remove C-style cast.